content/en/docs/Responses/Child.md

```diff
@@ -100,11 +177,41 @@ description: >
 | `originalWidth` | `int` | No |     | The video original Width |
 | `originalHeight` | `int` | No |     | The video original Height |
 | `played` | `string` | No | **Yes**    | Date the album was last played. [ISO 8601]|
+| `bpm` | `int` | No |   **Yes**   | The BPM of the song. |
+| `comment` | `string` | No |  **Yes**    | The comment tag of the song. |
+| `sortName` | `string` | No |  **Yes**   | The song sort name. |
+| `musicBrainzId` | `string` | No |  **Yes**   | The track MusicBrainzID. |
```

Comment:
I just started implementing stuff so questions come just now...
Will we also need recording MBID? (used for example by listenbrainz for recommendations)

Reply:
This is the album MBid and it's exposed for the albums. I don't think there's a need to duplicate the data here since Subsonic have the scrobble endpoint. Else it's just a matter of requesting the album for the rare case it might be needed from a song no?

Reply:
Ah, you meant "release mbid" then. I would then suggest to rename the field to something more explicit, like `releaseMusicBrainzId`.
"track mbid" is used to identify a track within a release. "recording mbid" is used to identify a record, i.e some different tracks may actually point to the same recording (reeditions, compilations, etc.). Recording mbid is not only used for scrobling, but also to get recommendations. But all this is another story.

Reply:
Well yes you are on child so song, so the mbID is the track MB ID. With that you can have access to everything from MB database.

Reply:
I don't query MB myself since everything I need is already in tags, but it looks like there is no easy way to query relationships using track MBID (would require listing tracks in the associated release? see "Lookups" in https://musicbrainz.org/doc/MusicBrainz_API + https://musicbrainz.org/doc/MusicBrainz_Entity)

Reply:
This is a confusion between the tag names and the MB names.

Reply:
Yes indeed, that's why it would be better to make it clear in our specs (and IMO recording MBID is more useful than track MBID)
(by the way, Child is also used when browsing directories. In that case Child can be a directory that represents an album (with isDir=true), which is really messy for a generic mbid field. But that's another problem)

Reply:
The question is more about what users tags and providers have, that what we think could be useful but no one use.
I base this work on what Kodi / Plex do and Kodi is currently by a very large margin the most advanced in this department.

The tags that they expect and report are:
- MUSICBRAINZ ARTIST ID
- MUSICBRAINZ ALBUM ARTIST ID
- MUSICBRAINZ ALBUM ID
- MUSICBRAINZ_TRACKID (Or for ID3 UFID with owner "http://musicbrainz.org" that is the track ID)

For the first 2 ones there's nothing to say.
For the trackID, as I said despite the name being track ID this is the recordingID
https://picard-docs.musicbrainz.org/downloads/MusicBrainz_Picard_Tag_Map.html

While we can enhance the doc later, the naming confusion was always here and users who use this are supposed to know, it's not really our job to document the tags.

For the album it's the release ID. Again we can update the docs later.

What is a lot more problematic is your remark about Child being used for other meanings. I completely missed that and it suxx because not only it can be album but it can also be artists.
And if we start to expose artistIds for songs we need to expose as arrays too and it start to get messy.

Else we need a way to differentiate the Child mediaType and expose a mediaType field that contains Track/Album/Artist so we can expose the corresponding MBId without a risk for client to not understand what kind of Id it refers to.

Reply:
Or we could just always explicitely write: `artistMbid`, `releaseMbid`, `trackMbid`, `recordingMbid` and just set the relevant ones when needed (we could even set them all for a Child that is a song)
recording MBID encoded as MUSICBRAINZ_TRACKID and track MBID encoded as MUSICBRAINZ_RELEASETRACKID (or whatever) should be only an internal server problem. The server is here to handle various encoded tags from various formats.

Reply:
I've just explained that it does not work a song have multiple artist so you can't return a single artistMbid we need to return arrays, but then arrays makes no sense when it's about an artist. And then we are missing the albumArtistMdIds that is also an array and should be returned too. But those can already be returned in the corresponding artists field on the songs so duplicate values for no gain.

Furthermore the basic rule of OpenSubsonic about those additional fields is that if they are supported they must be returned so a Child artist would have the other fields returned but empty, without anyway for the client to still be sure about what the Child is about anyway. So more confusion for no gain.

Then having trackMbid and recordingMbid / releaseMbid would bring back the naming confusion on our side. Since we expose values that does not correspond anymore to tag names and what all other providers do but the MB names that most people are not aware of at all.

Even the release Mb Id is named album Id in Picard.

Our goal is to fix Subsonic API not MB/Picard and the mess in the naming they build. Let's be pragmatic and consistent with what every other media providers do and avoid adding unnecessary confusion to something already confusing.

If the current proposal (With addition of the mediaType for child) does not work for you even if it works for all other providers then we remove the MB stuff from this and build another proposal just for that.

Reply:
I didn't say artistMbid is single valued. But indeed for a single Child artist this makes no sense to have an array.

I really don't understand why being explicit about mbid brings confusion, it is actually for me the exact opposite?

In picard, here is what we get in the UI:
![image](https://github.com/opensubsonic/open-subsonic-api/assets/8963963/89b6a85a-5d3a-47cf-913e-7d4ecbbb46f5)
Encoding these fields in messy/legacy named tags is really a problem that has nothing to do with our API.

Reply:
It is and you are the confirmation that you are also affected :) You mix actual tags as what is in the files with MB names and what Picard shows you.

Your screen shot show **Tag** and the name **MusicBrainz Release Id** and yet this does not exist most other software will show another **Tag** name for that. The denomination of Tag in this case is an arbitrary internal name decided by Picard. (And in other software too).

And then there's the actual tags whose name even depend on the file and tag format. But for the MB Release ID they are all named as "Album Id". So there's no tag with the name Release Id never anywhere. 

This project is OpenSubsonic and deals with media files and tags, not what Picard hide behind the naming it wants. Many people have no idea Picard even exists.

So again either you agree on handling MB ids as the tags and how other media providers do and it's current proposal, else we remove them from this proposal and do it as another proposal. 
There's no need for endless discussion here about difference between tags and names in software that are not related to this one.

Reply:
I really don't care myself, internally I already use the "recording_mbid" and "track_mbid" denominations (as ListenBrainz requires recording_mbid or track_mbid json fields, not the things picard translates in tags).

That being said, if you all want trackMbid be the recording MBID (and possibly trackReleaseMbid be the track MBID), just because this is the name in the tags, that's perfectly fine for me, as long as we write in the comment what is is actually is.

Reply:
Again this is not the proposal :) The exposed names are not trackMbId and whatever but **musicBrainzId**, same name for different entities that then map to what correspond the best in the internal naming of the tools or stuff you want to use.
We do not enter this endless naming debate on purpose because it's not our fight nor our concern.

The problem about our doc it that we will bring that debate again to expose all the case and tags and everything and document all the tags in each formats.

For the mapping I've already given https://picard-docs.musicbrainz.org/downloads/MusicBrainz_Picard_Tag_Map.html that shows all in a simpler way. But again this is a mapping that is internal and specific to Picard.

I use MP3Tag and https://docs.mp3tag.de/mapping/#musicbrainz_trackid the exposed name is MUSICBRAINZ_TRACKID and the concept of recording ID would mean absolutely nothing for the users of that software, it's the MB track ID for them.

And it's the same for many other tools. We do not care about the tools, we care about the tags, the tags have names that are easy to understand and commonly used. That's current proposal and it works for the other providers.

Reply:
Ok let's try to break this endless discussion.
* Are you happy with the current proposal or not?
* What is for you the musicBrainzId when Child is used for a song?

Reply:
I already tried to stop the discussion twice ;) The current proposal works and is what others do. If you want something else we postpone the MB stuff. Each time you avoided the decision.

The Child will have a new mediaType field to improve the case when it's used for something else than songs (This helps for many other things than just MB so is beneficial by itself).

When Child is a song (so a track) this is the track ID that is in Picard / MB world the recording ID but in all the rest of the world the track ID.

If you remove Picard from the equation and take MP3Tag as the source then this discussion would not even exist as only TRACKID and ALBUMID exist and perfectly match the API.

If you remove all from the equation and only take the tags
We have track ID tag = the id of the track = MB recording ID and release track Id = the release track Id = MB release track ID.

The whole world and the tags match this definition, but Picard have decided otherwise and wants to show the MB names instead of the tag names to add some more confusion.

If you remove Picard a MB track ID is well know, and an MB album ID is well known and maps to recording / release in MB. 

Picard have decided that he wanted to hide the tags and do something else and confuse people, they can, but it's not our problem.

Reply:
You are focused on the names of the tags and on the tools/softwares that blindly read and repeat them. You already know that I don't agree with this approach. That's no big deal, let's end this discussion and let's go with your proposal!
But please put in the associated comment that in case of a song, the musicBrainzID is actually the recording MBID (and not the track MBID as it is stated now).

Reply:
Again not my proposal, it's what all the others do and the names every one but Picard use. I offered you to build what you'd call a proper proposal but you do not seem to want to.

And I tried to explained I don't know how many times, that we are talking about software that read the tags and expose them ... Subsonic is a media provider that expose the media and their tags, they are not MusicBrainz clients that must use musicbrainz to have data to expose to clients.

I know my English is not that fluent but really maybe someone can jump in to explain...

99% of Subsonic servers that will expose the musicBrainzId will just blindly take the tag and expose it to the client.
For Songs they will take the tag that is named Track ID yes I know this is a stupid name but this is it's name not my call here just a fact ...

If the user actually put a release track ID there then 99% of the servers won't know, if a user put his mother name in that field then again 99% of the servers will never know.

So from there 2 solutions: We talk about tags and do not enforce anything, doc is correct. Else we enforce that the song musicBrainzId is really a recording ID and then it means that the server needs to validate the data and do calls to musicBrainz to ensure data integrity. Something I tried to explain would be a bad idea as we do not want to enforce such support on servers. Many servers would agree to blindly expose the tags, very few would add support for MB if they do not already support it. IMO having the data exposed without validation can still be a gain for some users and would be better than limiting the data to server that fully embrace MB. 

So yes this is an endless discussion, but this is an important one so that you understand the choices and agree with them, because that's the future of the API and servers that depends on those choices.

Anything enforced in the doc is enforcing, if we force tag data to match another website data, then servers must ensure data integrity and do the server calls.

In the same way we do not enforce anything in the sort name values, if a user want to put numbers in those tags to sort by a very own specific order it's his choice, and Subsonic will just blindly expose that data.

So TL;DR:
Do you understand why I talk about tags and not Picard / MusicBrainz?

And again: If you think only a proper MB integration is the way to go, then I remove this from this proposal and we build a proposal for MB support, something very different than just exposing a tag.

Reply:
Well I proposed trackMBID/recordingMBID/artistMBID/releaseMBID but you rejected it.
You want to stick to raw tags, fine. Let's go with your proposal, let's move on.

Reply:
So you do not understand nor agree. Your half proposal is tied to MB not tags, I have no problem moving to that if that's what is preferred ....

I do not want to stick to anything I try to explain that we either use tags or MB and that's it's then a different API and a different work for the servers. But we can't be in the middle with half baked stuff. MB Ids are very important on the server side for a few things, but the usage on the client side is a lot more limited.

I don't know how to explain this more and you seems too angry to try anymore to read what I wrote.

This is a team work, even if I feel that if I do not push nothing moves, but it's important that everyone agree else this won't bring anything.

I'll remove the MB support from this proposal and we can discuss a better proposal when heads are cooled off. Let's not block this for something nearly no server currently support anyway.

Reply:
Don't get me wrong: your proposal works and I (really) do agree to move on it :)

What I propose is just to use the MusicBrainz names for our fields (as after all, we do expose their stuff, and that's how they refer to their own stuff in their docs + in their APIs) instead of sticking to the messy tag names.

For me, putting a `recordingMBID` field instead of `musicBrainzId` field does not imply the server needs to validate its content to the MB database, but yes indeed this implies all servers will need to be aware that MUSICBRAINZ_TRACKID is actually a recording MBID, even if they don't care (but I don't even expect them to check it is a properly formatted UUID).

But again, no problem to go the raw tags way, **really**.

Reply:
on the topic of using the proper MB name vs the tag name, is that not all subsonic servers are based around music tags. for example the gazelle-subsonic, and beets-subsonic projects. they serve a subsonic compatible API, but are not based on tags (they're based on the gazelle API and beets.io library, respectively. and they have explicit distinctions on release vs recording, release vs release group, etc)

Reply:
The thing is that if they can do more they can do less :) The point is really about not enforcing things on servers that can't handle it while still allowing raw tag passing from servers to client that may want it.

In the same vein using the generic mbId name makes it non blocking for any future update that could expose all the MB data as the MB names are non clashing with this.

So let's go with this for now, I'll add the mediaType to child after the wait delay to avoid restarting again a round of votes.

There's still room to build something specific for MB the same way that subsonic API currently have hardcoded stuff for lastFM.